app/controllers/katello/api/rhsm/candlepin_proxies_controller.rb

```diff
@@ -381,13 +381,7 @@ def get_content_view_environments(label = nil, organization = nil)
           where("#{Organization.table_name}.id = ?", organization.id)
       environments = environments.where("#{Katello::ContentViewEnvironment.table_name}.label = ?", label) if label
 
-      environments.delete_if do |env|
-        if env.content_view.default
-          !env.environment.readable?
-        else
-          !env.content_view.readable?
-        end
-      end
+      environments = environments.to_a.delete_if { |env| (env.content_view.default && !env.environment.readable?) || !env.content_view.readable? }
```

Comment:
@daviddavis 'reject' is what I had before and was told to change it, I am really cool with using  ```to_a.delete_if``` or ```reject``` and from what I can see they do the same thing and both have to be assigned back to a variable, is there an advantage to one or the other?